src/Tracy/Dumper.php

```diff
@@ -415,26 +415,31 @@ public static function encodeString($s, $maxLength = NULL)
 			$table["\t"] = '\t';
 		}
 
-		if (preg_match('#[^\x09\x0A\x0D\x20-\x7E\xA0-\x{10FFFF}]#u', $s) || preg_last_error()) {
+		if (function_exists('mb_substr')) {
+			if ($maxLength && strlen($s) > $maxLength) {
+				$s = mb_substr($tmp = $s, 0, $maxLength, 'UTF-8');
+				$shortened = $s !== $tmp;
+			} else {
+				$shortened = false;
+			}
+			if (preg_match('#[^\x09\x0A\x0D\x20-\x7E\xA0-\x{10FFFF}]#u', $s) || preg_last_error()) {
+				$s = strtr($s, $table);
+			}
+		} elseif (preg_match('#[^\x09\x0A\x0D\x20-\x7E\xA0-\x{10FFFF}]#u', $s) || preg_last_error()) {
 			if ($shortened = ($maxLength && strlen($s) > $maxLength)) {
 				$s = substr($s, 0, $maxLength);
 			}
 			$s = strtr($s, $table);
```

Comment:
After changing the elseif statement back to if, you wil need to change the `$shortened` variable only the string is futher, i.e.

~~~php
if ($maxLength && strlen($s) > $maxLength) {
	$s = substr($s, 0, $maxLength);
	$shortened = TRUE;
}
$s = strtr($s, $table);
~~~